pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -640,15 +640,20 @@ private static String database(Properties props) {
    * @return the timeout from the URL, in milliseconds
    */
   private static long timeout(Properties props) {
+    int result = 0;
     String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
     if (timeout != null) {
       try {
-        return (long) (Float.parseFloat(timeout) * 1000);
+        result = Integer.parseInt(timeout);
```

Comment:
As far as I understand, the only reason is "I assume loginTimeout value should always be integer in secs".

There might be even a good reason to use float values: ability to specify sub-second timeouts while still being in line with DriverManager's `second` units.

Having said that, I thing we should refrain from altering the code.